zapread.com/Controllers/GroupController.cs

```diff
@@ -1113,20 +1113,40 @@ public PartialViewResult GetGroupTags(int groupId)
         [HttpPost]
         [ValidateJsonAntiForgeryToken]
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA3147:Mark Verb Handlers With Validate Antiforgery Token", Justification = "<Pending>")]
+        [Route("Group/Icon/Update")]
         public ActionResult UpdateGroupIcon(int groupId, string icon)
         {
+            // calling user id
+            var userAppId = User.Identity.GetUserId();
+
+            if (userAppId == null)
+            {
+                Response.StatusCode = (int)HttpStatusCode.Unauthorized;
+                return Json(new { success = false, message = "User not authorized."});
+            }
+
             using (var db = new ZapContext())
             {
-                var g = db.Groups.Where(grp => grp.GroupId == groupId).FirstOrDefault();
+                var g = db.Groups.Where(grp => grp.GroupId == groupId)
+                    .Include(gr => gr.Administrators)
```

Comment:
You should add `.Include(gr => gr.Administrators)` to  `UpdateGrouptags` aswell.